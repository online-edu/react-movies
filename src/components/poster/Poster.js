import React from 'react';
import PropTypes from 'prop-types';
import './Poster.scss';

const Poster = ({ url, id, name, click }) => (
    <figure className="movie-poster d-flex justify-content-center">
        <img
            className="movie-poster__image d-flex"
            alt={name}
            src={`https://image.tmdb.org/t/p/w185_and_h278_bestv2${url}`}
            onClick={() => click(id)}
            onKeyPress={() => click(id)}
            role="presentation"
        />
    </figure>
);

Poster.propTypes = {
    url: PropTypes.string.isRequired,
    id: PropTypes.number,
    name: PropTypes.string.isRequired,
    click: PropTypes.func,
};

export default Poster;
