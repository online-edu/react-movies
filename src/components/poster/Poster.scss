.movie-poster {
    .movie-poster__image {
        cursor: pointer;
        border-radius: 12px;
        transition: all 0.3s ease-in;
        &:hover {
            transform: scale(1.2);
        }
    }
}
