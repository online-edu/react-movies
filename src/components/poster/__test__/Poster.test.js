import React from 'react';
import { shallow, mount } from 'enzyme';
import Poster from '../Poster';

describe('<Poster />', () => {
    const onPosterClick = jest.fn();
    const wrapper = (
        <Poster
            id={399579}
            name="Alita: Battle Angel"
            click={onPosterClick}
            url="xRWht48C2V8XNfzvPehyClOvDni.jpg"
        />
    );
    it('renders correctly', () => {
        const component = shallow(wrapper);
        expect(component).toMatchSnapshot();
    });

    it('contains  <img /> tag', () => {
        const component = shallow(wrapper);
        expect(component.find('img')).toHaveLength(1);
    });

    it('allows us to set props', () => {
        const component = mount(wrapper);
        expect(component.props().name).toEqual('Alita: Battle Angel');
        component.setProps({ name: 'Creed II' });
        expect(component.props().name).toEqual('Creed II');
    });

    it('simulates click/keydown events', () => {
        const component = shallow(wrapper);
        component.find('button').simulate('click');
        component.find('button').simulate('keyPress');
        expect(onPosterClick).toHaveBeenCalledTimes(2);
    });
});
