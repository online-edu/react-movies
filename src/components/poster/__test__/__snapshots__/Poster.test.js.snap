// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Poster /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Poster
    click={[MockFunction]}
    id={399579}
    name="Alita: Battle Angel"
    tabIndex={1}
    url="xRWht48C2V8XNfzvPehyClOvDni.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="Alita: Battle Angel"
        className="movie-poster__image d-flex rounded"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="presentation"
        src="https://image.tmdb.org/t/p/w185_and_h278_bestv2xRWht48C2V8XNfzvPehyClOvDni.jpg"
        tabIndex={1}
      />,
      "className": "movie-poster d-flex justify-content-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Alita: Battle Angel",
        "className": "movie-poster__image d-flex rounded",
        "onClick": [Function],
        "onKeyPress": [Function],
        "role": "presentation",
        "src": "https://image.tmdb.org/t/p/w185_and_h278_bestv2xRWht48C2V8XNfzvPehyClOvDni.jpg",
        "tabIndex": 1,
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "figure",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="Alita: Battle Angel"
          className="movie-poster__image d-flex rounded"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="presentation"
          src="https://image.tmdb.org/t/p/w185_and_h278_bestv2xRWht48C2V8XNfzvPehyClOvDni.jpg"
          tabIndex={1}
        />,
        "className": "movie-poster d-flex justify-content-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Alita: Battle Angel",
          "className": "movie-poster__image d-flex rounded",
          "onClick": [Function],
          "onKeyPress": [Function],
          "role": "presentation",
          "src": "https://image.tmdb.org/t/p/w185_and_h278_bestv2xRWht48C2V8XNfzvPehyClOvDni.jpg",
          "tabIndex": 1,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "figure",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
